#166272358 Read History

EMacco opened the pull request:
#### What does this PR do?
Store and fetch read history for a logged in user.

#### Description of Task to be completed?
- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user `GET /users/history`
- Document swagger

#### How should this be manually tested?
- npm install 
- Login `POST /auth/login`
- Call endpoint to read a particular article `GET /articles/:slug`
- Fetch your history `GET /users/history`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#166272358](https://www.pivotaltracker.com/story/show/166272358)

#### Screenshots (if appropriate)
<img width="656" alt="Screenshot 2019-06-24 at 11 59 23 AM" src="https://user-images.githubusercontent.com/20377428/60014920-112f4d80-967a-11e9-9700-38fa2afde083.png">

#### Questions:
N/A

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

adex001 reviewed (dismissed):
Nice job..

emmygozi reviewed (dismissed):
Looks good implementing feedback and syncing with team members

fidelisojeah commented on server/helpers/Token.js:
```diff
@@ -76,5 +76,38 @@ class Token {
+   * @param {*} next
+   * @return {null} Returns nothing
+   */
+  static async authenticate(req, res, next) {
```
Why is this here?

Is it any part of your ticket?

EMacco replied:
Yes. 
My ticket requires that I authenticate a user if the user is signed in so that I will be able to log the article into the user's read history

If I used the authorize middleware, it will not allow a logged out user to access the route, so I created the authentication middleware so that it identifies the user is he is logged in, but ignores if the user is not.

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]

EMacco committed:
feat-[166272358]: Read History

- Write tests
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user
- Document swagger

[Finishes #166272358]
